components/user_groups_modal/user_groups_modal_header/user_groups_modal_header.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+
+import {Modal} from 'react-bootstrap';
+
+import {FormattedMessage} from 'react-intl';
+
+import CreateUserGroupsModal from 'components/create_user_groups_modal';
+import {ModalData} from 'types/actions';
+import {ModalIdentifiers} from 'utils/constants';
+
+export type Props = {
+    canCreateCustomGroups: boolean;
+    onExited: () => void;
+    backButtonAction: () => void;
+    actions: {
+        openModal: <P>(modalData: ModalData<P>) => void;
+    };
+}
+
+const UserGroupsModalHeader = (props: Props) => {
+    const goToCreateModal = () => {
```

Comment:
This is created for every single render, so it is going to force the re-render of the underneath component in every render too. a `useCallback` can help.